setup/Setup.CLI/Commands/PatchSubCommands.cs

```diff
@@ -0,0 +1,62 @@
+using Cocona;
+using Cocona.Application;
+using DiffPatch;
+using Terraria.ModLoader.Setup.Core;
+
+namespace Terraria.ModLoader.Setup.CLI.Commands;
+
+public sealed class PatchSubCommands : BaseCommand
+{
+	private readonly PatchTask.Factory patchTaskFactory;
+
+	public PatchSubCommands(
+		PatchTask.Factory patchTaskFactory,
+		ProgramSettings programSettings,
+		TargetsFilesUpdater targetsFilesUpdater,
+		TerrariaExecutableSetter terrariaExecutableSetter)
+		: base(programSettings, targetsFilesUpdater, terrariaExecutableSetter)
+	{
+		this.patchTaskFactory = patchTaskFactory;
+	}
+
+	[Command("terraria", Description = "Patches source in src/Terraria")]
+	public async Task PatchTerraria(PatchParameters parameters, [FromService] ICoconaAppContextAccessor contextAccessor)
+	{
+		PatchTask patchTask = GetPatchTask(PatchTaskParameters.ForTerraria(ProgramSettings), parameters.PatchMode);
+
+		await RunTask(patchTask, parameters.PlainProgress, cancellationToken: contextAccessor.Current?.CancellationToken ?? default);
+	}
+
+	[Command("terraria-net-core", Description = "Patches source in src/TerrariaNetCore")]
+	public async Task PatchTerrariaNetCore(PatchParameters parameters, [FromService] ICoconaAppContextAccessor contextAccessor)
+	{
+		PatchTask patchTask = GetPatchTask(PatchTaskParameters.ForTerrariaNetCore(ProgramSettings), parameters.PatchMode);
+
+		await RunTask(patchTask, parameters.PlainProgress, cancellationToken: contextAccessor.Current?.CancellationToken ?? default);
+	}
+
+	[Command("tml", Description = "Patches source in src/tModLoader. Edit the source code in src/tModLoader after this phase. Internally formats the sources before patching.")]
+	public async Task PatchTML(PatchParameters parameters, [FromService] ICoconaAppContextAccessor contextAccessor)
+	{
+		PatchTask patchTask = GetPatchTask(PatchTaskParameters.ForTModLoader(ProgramSettings), parameters.PatchMode);
+
+		await RunTask(patchTask, parameters.PlainProgress, cancellationToken: contextAccessor.Current?.CancellationToken ?? default);
+	}
+
+	private PatchTask GetPatchTask(PatchTaskParameters parameters, Patcher.Mode? patchMode)
+	{
+		if (patchMode.HasValue && patchMode != ProgramSettings.PatchMode) {
+			ProgramSettings.PatchMode = patchMode.Value;
+			ProgramSettings.Save();
+		}
+
+		return patchTaskFactory(parameters);
+	}
+}
+
+public record PatchParameters : BaseParameters
+{
+	[Option('m', Description = "Set the patch mode. Default is EXACT. If you change the patch mode, it will be saved and can be omitted for further calls.")]
+	[HasDefaultValue]
+	public Patcher.Mode? PatchMode { get; set; }
```

Comment:
I know the default is for this flag to be sticky in GUI, but it's likely that CLI users will not read this explanation in the help text, and that having an argument to a command be 'sticky' will contradict their intuitions. 

If we do want a way to apply this setting permanently via CLI, I recommend either a separate command, or a separate argument (eg `--set-patch-mode`)

Reply:
Changed. The save calls were more or less based on a previous state of the application where I always had to save the configuration.